test/as_documentation/metaconstants.clj

```diff
@@ -145,22 +145,35 @@
     (provided
       --mc-- =contains=> {:c 50000})))
 
+(unfinished gen-doc)
+(fact "Test merging of metaconstant that appear in data and function fakes"
+  (:header (gen-doc)) => "gamma"
+  (provided
+    (gen-doc) => ..doc..
+    ..doc.. =contains=> {:header "gamma"}))
 
+(fact "Test merging of nested metaconstant that appear in data and function fakes"
+  (:header (:raw (gen-doc))) => "gamma"
+  (provided
+    (gen-doc) => {:raw ..doc..}
+    ..doc.. =contains=> {:header "gamma"}))
```

Comment:
Would the following maintain the same behavior as now?
```
(fact 
  (identical? (gen-doc) (gen-doc)) => true
  (provided
    ..doc.. =contains=> {:header "gamma"}
    (gen-doc) => ..doc..)) 
```

What about this:
```
(fact 
  (identical? (gen-doc) (gen-doc)) => true
  (provided
    ..doc.. =contains=> {:header (Date.)}
    (gen-doc) => ..doc..)) 
```
?